pkg/server: TestInitStatsSessionBlockGC is affected by global stats lease left by previous tests

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Run the following in `pkg/server` with failpoint enabled:

```bash
make failpoint-enable && (
  cd pkg/server &&
  go test -v -run 'TestCursorExistsFlag|TestInitStatsSessionBlockGC' -tags=intest,deadlock
); rc=$?; make failpoint-disable; exit $rc
```

`TestCursorExistsFlag` passes, but `TestInitStatsSessionBlockGC` fails with:

```text
Messages: min_start_ts is not blocked over 1s
```

A smaller reproducer is any test that uses `testkit.CreateMockStoreAndDomain()` before `TestInitStatsSessionBlockGC` in the same test binary.

### 2. What did you expect to see? (Required)

`TestInitStatsSessionBlockGC` should not depend on the process-global `stats lease` value left behind by previous tests. It should pass regardless of earlier tests in the same package.

### 3. What did you see instead (Required)

`testkit.CreateMockStoreAndDomain()` calls `session.DisableStats4Test()`, which sets the process-global stats lease to `-1` and does not restore it. When `TestInitStatsSessionBlockGC` runs afterward, `session.BootstrapSession(store)` sees `stats lease = -1ns`, so the internal stats init path no longer blocks GC as the test expects.

Observed logs from the reproducer:

```text
=== RUN   TestCursorExistsFlag
... ["stats lease"=-1ns]
--- PASS: TestCursorExistsFlag
=== RUN   TestInitStatsSessionBlockGC
... ["stats lease"=-1ns]
Messages: min_start_ts is not blocked over 1s
--- FAIL: TestInitStatsSessionBlockGC
```

### 4. What is your TiDB version? (Required)

Reproduced on `master` at commit `3b531d4e3d`.

--- a/pkg/server/stat_test.go
+++ b/pkg/server/stat_test.go
@@ -25,6 +25,7 @@ import (
 	"github.com/pingcap/tidb/pkg/keyspace"
 	"github.com/pingcap/tidb/pkg/server/internal/util"
 	"github.com/pingcap/tidb/pkg/session"
+	"github.com/pingcap/tidb/pkg/sessionctx/vardef"
 	"github.com/pingcap/tidb/pkg/store/mockstore"
 	"github.com/stretchr/testify/require"
 	"github.com/tikv/client-go/v2/oracle"
@@ -71,6 +72,9 @@ func TestInitStatsSessionBlockGC(t *testing.T) {
 	defer func() {
 		config.StoreGlobalConfig(origConfig)
 	}()
+	origStatsLease := vardef.GetStatsLease()
+	defer vardef.SetStatsLease(origStatsLease)
+	vardef.SetStatsLease(3 * time.Second)
 	newConfig := *origConfig
 	for _, lite := range []bool{false, true} {
 		require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/statistics/handle/beforeInitStats", "pause"))

